src/proto/streams/streams.rs

```diff
@@ -866,6 +866,67 @@ impl<B> StreamRef<B> {
         })
     }
 
+    pub fn send_push_promise(
+        &mut self,
+        request: Request<()>,
+    ) -> Result<StreamRef<B>, UserError> {
+        let mut me = self.opaque.inner.lock().unwrap();
+        let me = &mut *me;
+
+        let mut send_buffer = self.send_buffer.inner.lock().unwrap();
+        let send_buffer = &mut *send_buffer;
+
+        let promised_id = {
+            let actions = &mut me.actions;
```

Comment:
this seems weird --- is the subscope really necessary? it seems like you should be able to just say
```rust
let send_buffer = ...

let actions = &mut me.actions;
actions.send.reserve_local()?;

let child_key = ...
```
unless I'm missing something?

Reply:
probably was necessary at some point, fixed it